### PR TITLE
Feature/217556 delete future meals on settlement

### PR DESCRIPTION
**Pull request by sourcesoldier**

### CODE REVIEW

File: src/Mealz/AccountingBundle/Event/ProfileSettlementEvent.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+namespace App\Mealz\AccountingBundle\Event;
```
Comment (tgietzen) [inline src/Mealz/AccountingBundle/Event/ProfileSettlementEvent.php:2]: declare(strict_types=1); is missing (same for other new files).

File: src/Mealz/AccountingBundle/Controller/CostSheetController.php [hunk 2]
```diff
@@ -166,6 +169,15 @@ public function confirmSettlement(
             $profile = $queryResult[0];
             $profile->setSettlementHash(null);
 
+            // Dispatch profile.settlement.confirm event
+            $dispatcher = new EventDispatcher();
+
+            $subscriber = new SettlementSubscriber($this->getDoctrine()->getRepository('MealzMealBundle:Participant'));
```
Comment (chetan-thapliyal) [inline src/Mealz/AccountingBundle/Controller/CostSheetController.php:175]: Is there a reason why we are explicitly subscribing a subscriber where we are dispatching the event?

It should work with it.

File: src/Mealz/MealBundle/Entity/ParticipantRepository.php [hunk 3]
```diff
@@ -354,4 +354,33 @@ public function getPendingParticipants(DateTime $dateTime)
         return $queryBuilder->getQuery()
             ->getResult();
     }
+
+    /**
+     * Removes all future ordered meals for a given profile
+     *
+     * @param Profile $profile
+     */
+    public function removeFutureMealsByProfile(Profile $profile) {
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Entity/ParticipantRepository.php:363]: Symfony convention puts opening bracket on its own line.

File: src/Mealz/AccountingBundle/Event/Subscriber/SettlementSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+use App\Mealz\AccountingBundle\Event\ProfileSettlementEvent;
+use App\Mealz\MealBundle\Entity\ParticipantRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class SettlementSubscriber implements EventSubscriberInterface
+{
+
+    private ParticipantRepository $participantRepo;
+
+    public function __construct(ParticipantRepository $participantRepo){
```
Comment (chetan-thapliyal) [inline src/Mealz/AccountingBundle/Event/Subscriber/SettlementSubscriber.php:14]: Opening brace should be on new line.

File: src/Mealz/AccountingBundle/Event/Subscriber/SettlementSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace App\Mealz\AccountingBundle\Event\Subscriber;
+
+use App\Mealz\AccountingBundle\Event\ProfileSettlementEvent;
+use App\Mealz\MealBundle\Entity\ParticipantRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class SettlementSubscriber implements EventSubscriberInterface
+{
+
```
Comment (chetan-thapliyal) [inline src/Mealz/AccountingBundle/Event/Subscriber/SettlementSubscriber.php:11]: Empty line

File: src/Mealz/AccountingBundle/Controller/CostSheetController.php (hunk 2 above)
Comment (sourcesoldier) [reply src/Mealz/AccountingBundle/Controller/CostSheetController.php:175]: https://symfony.com/doc/current/components/event_dispatcher.html#using-event-subscribers This is what I've got from the doc.

Comment (chetan-thapliyal) [reply src/Mealz/AccountingBundle/Controller/CostSheetController.php:175]: This is another reference. https://stackoverflow.com/questions/57540654/symfony-4-custom-event-dispatcher-not-working

Why not write less code and let Symfony handle as many things as possible.

Comment (sourcesoldier) [reply src/Mealz/AccountingBundle/Controller/CostSheetController.php:175]: Yes, but the example you posted uses a LISTENER and I thought we would like to prefer the SUBSCRIBER method?

Comment (chetan-thapliyal) [reply src/Mealz/AccountingBundle/Controller/CostSheetController.php:175]: It is a  subscriber. 

`class TestSubscriber implements EventSubscriberInterface`

**Pull request merged**